### ISSUE TITLE
parts generator should show error/warning if non-component is in selection

### ISSUE THREAD

**Issue by josh-m-sharpe** [lairdubois/lairdubois-opencutlist-sketchup-extension#450]
Request:
OpenCutList should show an error if an object that isn't a component is part of the selection it is generating parts for.   It seems to do this if there are ZERO components in the selection, but not if there is 1 or more non-component object.

Context - I copied some plans off the internet.  I'm not a SketchUp Expert, but when I build things myself I always triple click all the things to make them Components before using OpenCutList.  To date this has worked great.  Something about this plan I copied off the internetz is weird.  The parts, I think are "grouped" and as such aren't components? So, after modifying the plans (and creating a few components along the way, when I went to use open cut list it shorted me.  Basically, it didn't see the "grouped" items and there was no warning at all.  It generated plans but I didn't look close enough to notice that a whole board was missing.  

As such I went to the lumber yard and didn't buy enough lumber - so now I have to go back. :) 

This is what OCL shows if I select the one item in question and then attempt to generate plans.  The request is that an error like this would show up if I have selected my entire object (or nothing at all) and OCL is generating a full set of plans.  In that case, no warning is shown, and it is not obvious at all that a component/item/grouped-thing is being skipped over.

<img width="1026" alt="Cursor_and_sketch" src="https://user-images.githubusercontent.com/39473/208310889-4c978b58-4fc4-4a78-be7b-d5cd133ee505.png">

Thank you for the excellent plugin!  Also, pretty neat that it's written in ruby! Cheers

**Comment by bbeaulant** [comment]
Hi Josh,

OCL can't generate this kind of warning. Because it ignores groups during exploration process then it doesn't enter in and then it can't count drawing into them.

I understand that in your sample it could be simple to have this kind of warning. But in a lot of other models, groups exists and they are group because we don't want to include them in part list. Then this kind of warning could be displayed quite every time and consume computation time for nothing on large models.

In you case, the best thing to do is to use the "Highlight parts in model" feature to see if OpenCutList catch everything.
It paint all of the parts in red.
Of course it would be better in this case to ha a model with a different color than red ...

<img width="551" alt="Capture d’écran 2022-12-18 à 18 33 55" src="https://user-images.githubusercontent.com/156056/208311369-58e67287-5c86-4df3-9554-99922fe87737.png">

**Comment by josh-m-sharpe** [comment]
Could it not count the number of items selected overall and compare that to the number of items it is going to process?

**Comment by bbeaulant** [comment]
No, because of the tree structure of SketchUp models, you can select on object that contains hundred of sub parts :)
